### PR TITLE
feat(node_monitoring): provide node monitoring guide

### LINKED ISSUE #32: Add Node Monitoring Guide
Provide easy-to-use Prometheus, Substrate Exporter, Node Exporter, Grafana integration.

### PR DESCRIPTION
**Pull request by SimonKraus**
resolves #32

asked @vgantchev to proofread

### CODE REVIEW

File: docs/node_monitoring.md [hunk 1]
```diff
@@ -0,0 +1,349 @@
+---
+id: node_monitoring
+title: Node Monitoring
+---
+
+import useBaseUrl from '@docusaurus/useBaseUrl'; 
+
+In this chapter we will walk you through setting up a local monitoring for your validator node.
```
Comment (vgantchev) [inline docs/node_monitoring.md:8]:
```suggestion
In this chapter we will walk you through the setup of local monitoring for your validator node.
```

File: docs/node_monitoring.md [hunk 2]
```diff
@@ -0,0 +1,349 @@
...
+You must have your [validator node](/node_setup) up and running.  
+This guide was tested on Ubuntu 20.04 LTS release.
+
+## Prometheus Setup
+
+In the first step we will set up the Prometheus server.
+
+### User and Directories
+
+We create a user just for the monitoring purpose which has no home directory and can't be used to log in.
```
Comment (vgantchev) [inline docs/node_monitoring.md:21]:
```suggestion
We create a user just for monitoring purposes which has no home directory and can't be used to log in.
```

File: docs/node_monitoring.md [hunk 3]
```diff
@@ -0,0 +1,349 @@
...
+
+```shell script
+$ sudo useradd --no-create-home --shell /usr/sbin/nologin prometheus
+```
+
+Then we create directories for the executable and configuration file.
```
Comment (vgantchev) [inline docs/node_monitoring.md:27]:
```suggestion
Then we create directories for the executable and the configuration file.
```

File: docs/node_monitoring.md [hunk 4]
```diff
@@ -0,0 +1,349 @@
...
+```
+
+We now need to assign those binaries to our freshly created user.
+
+```shell script
+$ sudo chown prometheus:prometheus /usr/local/bin/prometheus
+$ sudo chown prometheus:prometheus /usr/local/bin/promtool
+```
+
+Next up we'll copy over the web interface and configuration presets.
```
Comment (vgantchev) [inline docs/node_monitoring.md:71]:
```suggestion
Next up we'll copy the web interface and the configuration presets.
```

File: docs/node_monitoring.md [hunk 5]
```diff
@@ -0,0 +1,349 @@
...
+```
+
+You may have guessed it already but we're also changing the ownership of those directories.
+
+```shell script
+$ sudo chown -R prometheus:prometheus /etc/prometheus/consoles
+$ sudo chown -R prometheus:prometheus /etc/prometheus/console_libraries
+```
+
+We now got everything we need from the downloaded package so we will go one step back and do some cleanup.
```
Comment (vgantchev) [inline docs/node_monitoring.md:85]:
```suggestion
We now have everything we need from the downloaded package so we will go one step back and do some cleanup.
```

File: docs/node_monitoring.md [hunk 6]
```diff
@@ -0,0 +1,349 @@
...
+    scrape_interval: 5s
+    static_configs:
+      - targets: ["localhost:9090"]
+  - job_name: "substrate_node"
+    scrape_interval: 5s
+    static_configs:
+      - targets: ["localhost:9615"]
+```
+
+The first scrape job exports data of Prometheus itself, the second one carries the HydraDX node metrics.
```
Comment (vgantchev) [inline docs/node_monitoring.md:124]:
```suggestion
The first scrape job exports data of Prometheus itself, the second one exports the HydraDX node metrics.
```

File: docs/node_monitoring.md [hunk 7]
```diff
@@ -0,0 +1,349 @@
...
+We adjusted the `scrape_interval` of both jobs to get a more detailed statistic. This overrides the global values.
```
Comment (vgantchev) [inline docs/node_monitoring.md:125]:
```suggestion
We adjusted the `scrape_interval` of both jobs to get more detailed statistics. This overrides the global values.
```

File: docs/node_monitoring.md [hunk 8]
```diff
@@ -0,0 +1,349 @@
...
+  --storage.tsdb.path /var/lib/prometheus/ \
+  --web.console.templates=/etc/prometheus/consoles \
+  --web.console.libraries=/etc/prometheus/console_libraries
+  ExecReload=/bin/kill -HUP $MAINPID
+
+[Install]
+  WantedBy=multi-user.target
+```
+
+We will do three steps in one now but here's what it does:
```
Comment (vgantchev) [inline docs/node_monitoring.md:166]:
```suggestion
Next we will perform the following three steps:
```

File: docs/node_monitoring.md [hunk 9]
```diff
@@ -0,0 +1,349 @@
...
+`systemctl deamon-reload` loads new configurations and updates existing 
+`systemctl enable` activates our new service
+`systemctl start` triggers the execution of the service
+
+Please perform all three actions in one step now:
```
Comment (vgantchev) [inline docs/node_monitoring.md:171]:
```suggestion
You can perform the steps above in one command by executing:
```

File: docs/node_monitoring.md [hunk 10]
```diff
@@ -0,0 +1,349 @@
...
+Password: `admin`  
+
+<div style={{textAlign: 'center'}}>
+  <img src={useBaseUrl('/node-monitoring/grafana-home.png')} />
+</div>
+
+### Configuring the Datasource
+
+Please click the settings gear in the menu and select datasources.  
+In the next window you click "Add Datasource" and celect "Prometheus".
```
Comment (vgantchev) [inline docs/node_monitoring.md:313]:
```suggestion
In the next window you click "Add Datasource" and select "Prometheus".  
```

**Review by vgantchev — COMMENTED** [review]
Great job @SimonKraus, the guide is awesome and super easy to follow 👏 🥇 

wdyt about placing this in our Knowledge Base, under a new sub-menu called Validators. This would be super useful also beyond Snakenet

**Review by vgantchev — APPROVED** [review]
lgtm, we can take care of navigation when we revisit the menu and file atructure

**Review by martinfridrich — APPROVED** [review]

File: docs/node_monitoring.md [hunk 11]
```diff
@@ -0,0 +1,348 @@
...
+And paste the following configuration into it.
+
+```
+[Unit]
+Description=Node Exporter
+Wants=network-online.target
+After=network-online.target
+
+[Service]
+User=node_exporter
```
Comment (vgantchev) [inline docs/node_monitoring.md:233]: should be `prometheus` 🙏

**Review by jak-pan — APPROVED** [review]

**Pull request merged**